packages/opentelemetry-plugin-http/src/http.ts

```diff
@@ -314,7 +320,20 @@ export class HttpPlugin extends BasePlugin<Http> {
           span.end();
           return returned;
         };
-        return original.apply(this, [event, ...args]);
+        try {
+          return original.apply(this, [event, ...args]);
+        } catch (error) {
+          const message = error.message;
+          span
+            .setAttributes({
+              [AttributeNames.HTTP_ERROR_NAME]: error.name,
+              [AttributeNames.HTTP_ERROR_MESSAGE]: message,
+            })
+            .setStatus({ code: CanonicalCode.UNKNOWN, message })
```

Comment:
Is this always going to be `UNKNOWN` status code? Does this apply for things like 500 errors, etc.? Or is this a problem around a connection failure?

Reply:
For this section, we won't know as we catch before making the real call.
When there is a call, we are listening several events and for example we do:
https://github.com/open-telemetry/opentelemetry-js/blob/82b411029dc2424af053cb1dad0d20a4095740a6/packages/opentelemetry-plugin-http/src/http.ts#L314

Reply:
See related tests that show spans on Error.